bittensor/_dataset/dataset_impl.py

```diff
@@ -1,678 +1,772 @@
+                        
         else:
-            logger.warning("Failed to get text".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-            self.IPFS_fails += 1
-            
-        return text 
-
-    def get_dataset(self , file_meta):
-        r""" Either load a dataset, which is a list of hashes, from disk or download it from IPFS
-        Args:
-            file_meta (dict of str: int):
-                Specify the details of the dataset in the format of {'Name': , 'Hash':}.
+            # Get the folder_hashes from the dataset.
+            folder_hashes = (await self.get_folder_hashes(self.dataset2hash[dataset]))[:self.max_directories]
+
+            # For each folder, get the text hashes.
+            tasks = []
+            for f in folder_hashes:
+                tasks.append(self.get_folder_text_hashes(f, dataset=dataset))
+
+            # Some hashes are incomplete, ensure they have Size and Hash Field.
+            for folder_text_file_metas in await asyncio.gather(*tasks):
+                for file_meta in folder_text_file_metas:
+                    if 'Size' in file_meta and 'Hash' in file_meta:
+                        hash2file_meta[file_meta['Hash']] = file_meta   
+                    
+            text_file_metas = list(hash2file_meta.values())
 
-        Return:
-            hashes (list):
-                The hashes from the dataset downloaded from disk or IPFS.     
-        """
-        # --- Load text from path
-        logger.success( f"Getting dataset: {file_meta['Name']}" )
+            # If save is true, then save the hashes into the {self.data_dir}/{dataset}/file_metas.
+            if save:
+                self.save_json(path=f'{dataset}/file_metas', obj=text_file_metas)
         
-        hashes = self.load_hash(file_meta)
+        # Calculate the size.
+        self.dataset_size_map[dataset]  = sum([fm['Size'] for fm in text_file_metas])
 
-        if hashes != None:
-            hashes = json.loads(hashes)
+        return text_file_metas
 
-        # --- If couldnt load from path, download text.
-        else:
-            response = self.get_ipfs_directory(self.dataset_dir, file_meta)
-            if (response != None) and (response.status_code == 200):
-                self.IPFS_fails = 0
-                hashes = response.json()
-
-                # --- Save text if the save_dataset flag is on.
-                if self.save_dataset :
-                    self.save_hash(file_meta, json.dumps(response.json()) )
-                    
-            else:
-                self.IPFS_fails += 1
-                logger.warning("Failed to get dataset".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-                return None
+    def set_data_size(self, batch_size:Optional[int] = None, block_size:Optional[int] = None, sequence_length:Optional[int] = None,  block_size_bytes:Optional[int]= None, buffer_size:Optional[int]=None) -> None:
```

Comment:
This line is too long. Generally, 80-120 characters no more. I find it easier to read if the arguments are indented on new lines.